version: '3.8'
services:
  postgres:
    image: postgres:14
    container_name: postgres_container
    environment:
      POSTGRES_USER: ${POSTGRES_USER}
      POSTGRES_PASSWORD: ${POSTGRES_PASSWORD}
      POSTGRES_DB: ${POSTGRES_DB}
    ports:
      - "5432:5432"
    volumes:
      - postgres_data:/var/lib/postgresql/data
    healthcheck:
      test: [ "CMD-SHELL", "pg_isready -U ${POSTGRES_USER}" ]
      interval: 5s
      timeout: 10s
      retries: 5

  python:
    build:
      context: .
    container_name: python_etl
    depends_on:
      postgres:
        condition: service_healthy
    environment:
      - DB_HOST=${POSTGRES_HOST}
      - DB_PORT=5432
      - DB_NAME=${POSTGRES_DB}
      - DB_USER=${POSTGRES_USER}
      - POSTGRES_PASSWORD=${POSTGRES_PASSWORD}
    volumes:
      - ./src:/app/src
    working_dir: /app
    command: sh -c "python src/main.py && tail -f /dev/null"

volumes:
  postgres_data:
